src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -36,12 +37,11 @@
      *
      * @param interval the genomic span covered by this shard
      * @param paddedInterval the span covered by this shard, plus any additional padding on each side (must contain the un-padded interval)
-     * @param readsSource source of reads from which to populate this shard
+     * @param readsSource source of reads from which to populate this shard; it could be null if no data source is available
```

Comment:
I'm happy with the implementation because reads are loaded in a lazy way, and that could be important in terms of performance. Nevertheless, my aim was to provide a base class for two classes of `WindowWalker`: one for reads and one for variants, but allowing to do others on demand for reference, other features, etc.

Thus, it is true that is wierd to iterate over `ReadShard` without reads, only needed to keep track of the `interval`/`paddedInterval` in a single classe. Thus, I'm planning to split the implementation in two to classes (`SlidingWindowReadWalker` and `SlidingWindowVariantWalker`), to keep abstractions simple.

By the way, I implemented for the moment the method that you asked for to keep track of the changes and comments.